File: Jardin De infantes/EE/Servicios/Encriptado/EncriptarSHA5.vb
Imports System.Security.Cryptography
Public Class EncriptarSHA5
    Inherits eeEncriptado
    Sub New(_ValorInicial As String)
        Me.valorInicial = _ValorInicial
        Me.ValorEncriptado = ""

    End Sub

    Public Overrides Function Hashear() As String

        Dim md5 As New MD5CryptoServiceProvider
        Dim bytValue() As Byte
        Dim bytHash() As Byte
        Dim i As Integer

        bytValue = System.Text.Encoding.UTF8.GetBytes(Me.valorInicial)

        bytHash = md5.ComputeHash(bytValue)
        md5.Clear()

        For i = 0 To bytHash.Length - 1
            Me.ValorEncriptado &= bytHash(i).ToString("x").PadLeft(2, "0")
        Next
        Return ValorEncriptado
    End Function
End Class

File: Jardin De infantes/Seguridad/EEseguridad/Encriptado/EncriptarSHA5.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Security.Cryptography;
namespace EEseguridad.Encriptado
{
 public   class EncriptarSHA5 : eeEnciptador
    {
        public EncriptarSHA5(string _valorInicial)
        {
            valorEncriptado = "";
            valorInicial = _valorInicial;
        }
        public override string Hashear()
        {
            MD5 md5 = System.Security.Cryptography.MD5.Create();

            byte[] inputBytes = System.Text.Encoding.ASCII.GetBytes(valorInicial);

            byte[] hash = md5.ComputeHash(inputBytes);
            StringBuilder sb = new StringBuilder();

            for (int i = 0; i < hash.Length; i++)

            {

                sb.Append(hash[i].ToString("X2"));

            }

            return sb.ToString();
          
        }
    }
}
